src/server/h1writer.rs

```diff
@@ -284,9 +292,18 @@ impl<T: AsyncWrite, H: 'static> Writer for H1Writer<T, H> {
     #[inline]
     fn poll_completed(&mut self, shutdown: bool) -> Poll<(), io::Error> {
         if !self.buffer.is_empty() {
-            let buf: &[u8] =
-                unsafe { &mut *(self.buffer.as_ref() as *const _ as *mut _) };
-            let written = self.write_data(buf)?;
+            let written = {
```

Comment:
It seems this change introduce some code duplication...

Technically getting `&self.buffer` is not a problem as we ain't going to access it in `write_data` and I guess that was the reason for this unsafe i.e. remove code duplication.

Reply:
Very mild code duplication, just the `if` statement. I think that's acceptable for the benefit of not having to do non-local reasoning in order to avoid memory unsafety.